pkg/operator/etcdendpointscontroller/etcdendpointscontroller.go

```diff
@@ -111,6 +112,40 @@ func (c *EtcdEndpointsController) syncConfigMap(ctx context.Context, recorder ev
 		}
 	} else if !errors.IsNotFound(err) {
 		klog.Warningf("required configmap %s/%s will be created because it was missing: %w", operatorclient.TargetNamespace, "etcd-endpoints", err)
+	} else if existing == nil {
+		// create endpoint addresses for each node
+		nodes, err := c.nodeLister.List(labels.Set{"node-role.kubernetes.io/master": ""}.AsSelector())
+		if err != nil {
+			return fmt.Errorf("unable to list expected etcd member nodes: %v", err)
+		}
+		endpointAddresses := map[string]string{}
+		for _, node := range nodes {
+			var nodeInternalIP string
+			for _, nodeAddress := range node.Status.Addresses {
+				if nodeAddress.Type == corev1.NodeInternalIP {
+					nodeInternalIP = nodeAddress.Address
+					break
+				}
+			}
+			if len(nodeInternalIP) == 0 {
+				return fmt.Errorf("unable to determine internal ip address for node %s", node.Name)
+			}
+			// the key is not important, the value is used to list etcd members with next sync and the key will be overwritten by the memberId on the next sync
+			endpointAddresses[node.Name] = nodeInternalIP
```

Comment:
I know what you mean here but the more I think about it, I'm not really in favor of creating the configmap with incorrect keys and then just relying on a resync to fix that up. It's just confusing to see this in tests and the expected revisions of the configmap.
I understand that the revert quick fix works but I'm hoping we can clean this up while we're addressing this bug.

I'm thinking we could potentially introduce method in the etcdcli wrapper that lets us create the client even when the configmap is not present.
https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/etcdcli/etcdcli.go#L107-L117
That way we won't need two separate ways (node list vs member list) of populating the configmap data.